File: clients/kotlin/src/main/kotlin/net/eternaltwin/client/EtwinClient.kt
package net.eternaltwin.client

import net.eternaltwin.auth.AuthContext
import net.eternaltwin.dinoparc.*
import net.eternaltwin.user.MaybeCompleteUser
import net.eternaltwin.user.UserId

interface EtwinClient {
  fun getSelf(auth: Auth): AuthContext

  fun getUser(auth: Auth, userId: UserId): MaybeCompleteUser

  fun getDinoparcUser(auth: Auth, options: GetDinoparcUserOptions): EtwinDinoparcUser

  fun getDinoparcDinoz(auth: Auth, options: GetDinoparcDinozOptions): EtwinDinoparcDinoz
}

File: clients/php/src/Client/EtwinClient.php
<?php declare(strict_types=1);

namespace Eternaltwin\Client;

use Eternaltwin\Auth\AccessTokenAuthContext;
use Eternaltwin\Auth\GuestAuthContext;
use Eternaltwin\Auth\UserAuthContext;
use Eternaltwin\User\User;
use Eternaltwin\User\UserId;

interface EtwinClient {
  /**
   * @param Auth $auth
   * @return AccessTokenAuthContext | GuestAuthContext | UserAuthContext
   */
  function getSelf(Auth $auth);

  function getUser(Auth $auth, UserId $userId): User;
}
